app/controllers/subscriptions_controller.rb

```diff
@@ -28,6 +28,15 @@ def create
     render json: { id: subscription.id }, status: status
   end
 
+  def change_frequency
```

Comment:
This endpoint and the endpoint above (`create`) are very similar. If a user "creates" a new subscription when they already have one, it is effectively the same as changing the frequency on an existing subscription. This works by "ending" the existing subscription and creating a new one with the `source` set as `:frequency_changed`. This allows us to have better historical analytics data.

I like the idea of a `PATCH` endpoint to make it more explicit that we're dealing with an existing subscription, and raising a 404 if the existing subscription doesn't exist, but I think it should work the same as the endpoint above: creating a new subscription record and ending the existing one.

Reply:
That sounds better, changed